### PR TITLE
refactor(web): update-verdict-button-in-dispute-overview

### LINKED ISSUE #1423: Case details: tweak "Check how the jury voted"
image

### Change
- If the user has been drawn in the current round
  - if the user has not voted yet rename "Check how the jury voted ->" into "Cast your vote ->".
  - else rename into "You have voted ->"

### Rationale
"Check how the jury voted" makes sense for someone who hasn't been drawn. But for someone who's been drawn, it's actually encouraging a "lazy juror" strategy. They should rather focus on making up their own mind on how to vote (by reviewing the evidence) + actually vote.

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
closes #1423
Added the following with :

1.  "You were drawn" : shown in evidence period to the juror if drawn
2. "Commit your vote": shown in commit period, if juror has not already committed
3. "You have committed": shown in commit period, if juror has already committed
4. "You failed to commit": show when juror failed to commit in commit period
5. "Cast your vote": shown in voting period, if juror hasn't already voted
6. "You have voted": shown if juror has already voted
7. "You failed to vote": shown if juror failed to vote during voting period

---

## PR-Codex overview
This PR focuses on introducing a `VotingContext` to manage voting-related data and logic. It includes the following changes:

- Added `VotingContextProvider` component
- Created `useVotingContext` hook to access voting context data
- Refactored code to use `useVotingContext` instead of individual hooks
- Updated imports and dependencies

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: web/src/components/Verdict/FinalDecision.tsx [hunk 1]
```diff
@@ -84,13 +87,23 @@ interface IFinalDecision {
...
   const { data: currentRulingArray } = useKlerosCoreCurrentRuling({ args: [BigInt(id ?? 0)], watch: true });
   const currentRuling = Number(currentRulingArray?.[0]);
   const answer = disputeTemplate?.answers?.[currentRuling! - 1];
+  const buttonText = useMemo(() => {
+    if (!wasDrawn || isDisconnected) return "Check how the jury voted";
+    if (disputeDetails?.dispute?.period === "evidence") return "You were drawn";
+    if (isCommitPeriod) return commited ? "You have committed" : "Commit your vote";
+    if (isVotingPeriod && !commited) return "You failed to commit";
+    if (isVotingPeriod && !hasVoted) return "Cast your vote";
+    return hasVoted ? "You have voted" : "You failed to vote";
```
Comment (jaybuidl) [inline web/src/components/Verdict/FinalDecision.tsx:105]:
```suggestion
    return hasVoted ? "You have voted" : "You have not appeared for juror duty";
```

**Review by jaybuidl — APPROVED** [review]

**Review by kemuru — DISMISSED** [review]
lgtm

File: web/src/components/Verdict/FinalDecision.tsx [hunk 2]
```diff
@@ -84,13 +87,23 @@ interface IFinalDecision {
...
   const { data: currentRulingArray } = useKlerosCoreCurrentRuling({ args: [BigInt(id ?? 0)], watch: true });
   const currentRuling = Number(currentRulingArray?.[0]);
   const answer = disputeTemplate?.answers?.[currentRuling! - 1];
+  const buttonText = useMemo(() => {
+    if (!wasDrawn || isDisconnected) return "Check how the jury voted";
+    if (disputeDetails?.dispute?.period === "evidence") return "You were drawn";
+    if (isCommitPeriod) return commited ? "You have committed" : "Commit your vote";
+    if (isVotingPeriod && !commited) return "You failed to commit";
+    if (isVotingPeriod && !hasVoted) return "Cast your vote";
+    return hasVoted ? "You have voted" : "You have not appeared for juror duty";
```
Comment (alcercu) [inline web/src/components/Verdict/FinalDecision.tsx:105]: I think we should re-imagine the texts 🤔  right now I feel like some of them are call to actions and some of them are informative, and that feels confusing.
I would have:
- `Cast your vote` in voting period and not `hiddenVotes`
- `Commit your vote` on commit period
- `Reveal your vote` on voting period with `hiddenVotes`
- leave it with `Check how the jury voted` for other cases

Other texts feel more like information, and I don't think it's intuitive to click them and go to the voting page.

(maybe we could change it to a call to action to check the evidence tab during the first evidence period)

**Review by alcercu — CHANGES_REQUESTED** [review]

File: web/src/components/Verdict/FinalDecision.tsx (hunk 2 above)
Comment (jaybuidl) [reply web/src/components/Verdict/FinalDecision.tsx:105]: Okay let's do that, more simple 👍  

No strong opinion about the CTA for the evidence tab, maybe follow up in a subsequent PR?

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**